llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -182,30 +174,23 @@ static bool expandLengthIntrinsic(CallInst *Orig) {
     Value *Mul = Builder.CreateFMul(Elt, Elt);
     Sum = Builder.CreateFAdd(Sum, Mul);
   }
-  Value *Result = Builder.CreateIntrinsic(
-      EltTy, Intrinsic::sqrt, ArrayRef<Value *>{Sum}, nullptr, "elt.sqrt");
-
-  Orig->replaceAllUsesWith(Result);
-  Orig->eraseFromParent();
-  return true;
+  return Builder.CreateIntrinsic(EltTy, Intrinsic::sqrt, ArrayRef<Value *>{Sum},
```

Comment:
Sometimes the createIntrinsic call itself is returned, and sometimes the Value * variable that was assigned to this call is returned instead. Should we stick to returning the variable?

Reply:
The only places the variable is returned are places where something else is done to it before replacing the old one or where it might come from different creation calls depending on the path taken. Defining a variable on one line just to return it on the next seems unnecessary